src/seedu/addressbook/data/person/UniquePersonList.java

```diff
@@ -122,6 +129,23 @@ public void remove(ReadOnlyPerson toRemove) throws PersonNotFoundException {
         }
     }
 
+    /**
+     * Favourites the equivalent person from the list.
+     *
+     * @throws PersonNotFoundException if no such person could be found in the list.
+     * @throws AlreadyFavouritedException if person is already favourited.
+     */
+    public void favourite(ReadOnlyPerson toFavourite) throws PersonNotFoundException, AlreadyFavouritedException {
+        if(internalList.get(internalList.indexOf(toFavourite)).getFavourite()) {
+            throw new AlreadyFavouritedException();
+        } else {
+            final boolean personFoundAndFavourited = internalList.get(internalList.indexOf(toFavourite)).setFavourite();
```

Comment:
The above violates Law of Demeter, which probably isn't taught yet.
That's why the better approach would be UniquePersonList telling target Person to make himself a favourite, and have the target Person be the one who throws the exception instead
Also, `internalList.get(internalList.indexOf(toFavourite))` is called a lot so refactoring it out would be good.
Also the smell from the setFavourite also return true has resurfaced here. personFoundAndFavourited  will always be true unless you run into a OOB Exception, which means that PersonNotFound Exception will never thrown and the program would malfunction.